commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -32,23 +34,27 @@
     private String coordsLog;
 
     protected static Set<String> categorySet;
+    private static List<String> categoryList;
```

Comment:
Yeah, I agree. I mistakenly attempted to keep static state for many of the inner classes and methods in MwVolleyApi, and consequently many of the variables and methods that they touched also had to be made static to avoid errors. I should have tried to refactor from the start. But there were issues with the refactoring that I couldn't quite sort out (discussed in pull request #36 ).